### ISSUE TITLE
ProcessText.queryTextActions is Null

### ISSUE THREAD

**Issue by ozayr** [flutter/flutter#144505]
### Steps to reproduce

upgraded flutter to latest stable, open a route/page that has some kind of text field or form element

### Expected results

do not throw exception

### Actual results

call stack 

```
DefaultProcessTextService.queryTextActions (/Users/ghost/dev/flutter/packages/flutter/lib/src/services/process_text.dart:122)
<asynchronous gap> (Unknown Source:0)
EditableTextState._initProcessTextActions (/Users/ghost/dev/flutter/packages/flutter/lib/src/widgets/editable_text.dart:2856)
<asynchronous gap> (Unknown Source:0)
```

```
 rawResults = await _processTextChannel.invokeMethod(
      'ProcessText.queryTextActions',
    ) as Map<Object?, Object?>;
```
      
on the above code getting the below error 
      
   
`type 'Null' is not a subtype of type 'Map<Object?, Object?>' in type cast`

if I just step through things work as normal

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Flutter Doctor output

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.19.2, on macOS 14.3.1 23D60 darwin-arm64, locale en-ZA)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.86.2)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!

**Comment by huycozy** [comment]
Hi @ozayr 
For us to further investigate this issue, please provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that doesn’t include 3rd party plugins or complex production code. Thank you!

**Comment by serikqaliev** [comment]
+1

**Comment by huycozy** [comment]
@serikqaliev Please share a reproduction example as mentioned above if any. Please [do not leave "+1", "me too" or "same here" comments](https://github.com/flutter/flutter/wiki/Issue-hygiene#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-bugs).

**Comment by McLavrushka** [comment]
+1

**Comment by GabrielGoliveira04** [comment]
Same problem

**Comment by greenisles** [comment]
This only started occurring after updating to latest Flutter. The file attached is from a clean/new project with a TextFormField widget added. At first I tried with one of my more complicated versions and then a super simplified version followed by the example provided at api.flutter.dev with the same result each time per the attached screen cap. 

[main.dart.zip](https://github.com/flutter/flutter/files/14488101/main.dart.zip)

[doc.txt](https://github.com/flutter/flutter/files/14488191/doc.txt)

![2024-03-05_12-05](https://github.com/flutter/flutter/assets/122420171/cd582ce6-8b92-4c06-8675-20a7c484febf)

**Comment by huycozy** [comment]
@greenisles I check the issue with below minimal sample code on Android app target but don't see the issue. Which platform and device (OS, model) are you checking on? 

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() async {
  runApp(
    MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: Center(
          child: TextFormField(
            decoration: const InputDecoration(
              icon: Icon(Icons.person),
              hintText: 'What do people call you?',
              labelText: 'Name *',
            ),
            onSaved: (String? value) {},
            validator: (String? value) {
              return (value != null && value.contains('@')) ? 'Do not use the @ char.' : null;
            },
          ),
        ),
      ),
    ),
  );
}

```
</details>

For anyone facing this issue, it would be helpful if you could share steps to reproduce and the entire crash stack trace in **text format** so that we have more information to move this forward. Thanks!

**Comment by greenisles** [comment]
> @greenisles I check the issue with below minimal sample code on Android app target but don't see the issue. Which platform and device (OS, model) are you checking on?
> Sample code
> 
> For anyone facing this issue, it would be helpful if you could share steps to reproduce and the entire crash stack trace in **text format** so that we have more information to move this forward. Thanks!

Hi, not sure if this helps but:
1. When running app from VSCode by using shortcut (F5) - the error is generated.
2. When running from VSCode terminal with command: flutter run (with or without --verbose switch) it does NOT error! Does this point to an error with the Flutter VSCode extension?

I'm currently using an iPhone 7 Plus sim running iOS 15.5 but then tried it on an iPhone 15 Pro Max sim with iOS 17.2 with the same error result.

**Comment by huycozy** [comment]
@greenisles I checked this on VSCode, ran the app in debug mode but still can't reproduce the issue. 

>Does this point to an error with the Flutter VSCode extension?

What are your VSCode and Flutter extension versions? When does the issue occur? (when you start entering the textfield or when the app starts?)

**Comment by greenisles** [comment]
@huycozy 

Version details below. In the provided app, the error occurs before the app is displayed. In my app, I'm displaying a modal dialog with two text fields; it errors before the dialog is displayed. If I continue (F5) after the error is displayed once for each text field, the form is displayed and the text widgets function as expected. 

Once the form is displayed; the text fields operate as expected with no further error. The error is always displayed when loading the form or any other form where the text field widget is being used.

**Flutter Extension: 3.84.0**
Published
2018-04-19, 04:49:03
Last released
2024-03-02, 01:55:24
Last updated
2024-03-02, 08:19:12

**VSCode:** 
Version: 1.87.0 (Universal)
Commit: 019f4d1419fbc8219a181fab7892ebccf7ee29a2
Date: 2024-02-27T23:42:56.944Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.3.0

**Comment by bleroux** [comment]
@greenisles
Do you have 'All exceptions' checked in VSCode 'Breakpoints' view?
![Capture d’écran 2024-03-05 à 12 01 11](https://github.com/flutter/flutter/assets/840911/797a8a7e-44a3-4962-9f86-49806eeb9808)

If yes, does the issue reproduced when unchecking 'All exceptions'?

If possible, thank you for trying on master channel because it was probably fixed by https://github.com/flutter/flutter/issues/141205.

**Comment by greenisles** [comment]
@bleroux 

Yes, I am breaking on all exceptions and yes, when disabled, the error is ignored. However, the reason I enabled this feature is I noticed some unusual route issues on closing the dialog; e.g. it was switching tabs unexpectedly, something that hadn't happened until the update.

I've switch to master channel and can confirm that the specific issue reported appears to have been resolved with #141205 so that's good news.

Cheers!

**Comment by huycozy** [comment]
Thank @bleroux for the spot and @greenisles for confirming this. I can reproduce the issue now on stable channel 3.19.2 with VSCode `All Exceptions` enabled:

![Screenshot 2024-03-06 at 11 09 56](https://github.com/flutter/flutter/assets/104349824/4f86ba94-1f76-4946-addd-0f3253071d50)

And it's indeed fixed on beta/master channels as https://github.com/flutter/flutter/pull/141205 has been rolled there.

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.19.2, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.19.2 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7482962148 (5 days ago), 2024-02-27 16:51:22 -0500
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] VS Code (version 1.87.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.84.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 122.0.6261.94

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.20.0-17.0.pre.94, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.20.0-17.0.pre.94 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a431dee5c9 (51 minutes ago), 2024-03-05 21:27:24 -0500
    • Engine revision effcf97a1f
    • Dart version 3.4.0 (build 3.4.0-190.0.dev)
    • DevTools version 2.33.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] VS Code (version 1.87.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.84.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.8 19H370
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 122.0.6261.94

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

For anyone facing this issue, please switch to beta/master channel. I'm going to close this as fixed. If you still see the issue on these channels, please write in comments. Thanks!
